waku/node/peer_manager/peer_manager.nim

```diff
@@ -121,8 +121,9 @@ proc addPeer*(pm: PeerManager, remotePeerInfo: RemotePeerInfo, origin = UnknownO
   discard remotePeerInfo.peerId.extractPublicKey(publicKey)
 
   if pm.peerStore[AddressBook][remotePeerInfo.peerId] == remotePeerInfo.addrs and
-     pm.peerStore[KeyBook][remotePeerInfo.peerId] == publicKey:
-    # Peer already managed
+     pm.peerStore[KeyBook][remotePeerInfo.peerId] == publicKey and
+     pm.peerStore[ENRBook][remotePeerInfo.peerId].raw.len > 0:
```

Comment:
Ah, could you add a comment (to the issue, not the code) on why this is necessary? Is this because these peers were explicitly connected to in addition to being added via discovery?